src/praisonai/tests/test_n8n_integration.py

```
@@ -499,13 +588,72 @@ def test_round_trip_conversion(self):
-        # Should have same number of agents (within reason - conversion may change names)
+        # Should have same number of agents
         original_agent_count = len(original_yaml["agents"])
         result_agent_count = len(result_yaml["agents"])
         assert result_agent_count == original_agent_count
+        
+        # Check that all steps are preserved
+        assert len(result_yaml["steps"]) == len(original_yaml["steps"])
+        
+        # Verify agent fields are preserved
+        result_agents = result_yaml["agents"]
+        for agent_id, original_agent in original_yaml["agents"].items():
```

Ruff (0.15.10)

[warning] Loop control variable `agent_id` not used within loop body

Rename unused `agent_id` to `_agent_id`

(B007)